pos_order_mgmt/static/src/js/widgets.js

```diff
@@ -0,0 +1,404 @@
+        },
+
+        action_print: function (order) {
+            var receipt = order.export_for_printing();
+            // We store temporarily the current order so we can safely compute
+            // taxes based on fiscal position
+            this.pos.current_order = this.pos.get_order()
+            this.pos.set_order(order);
+            if (this.pos.config.iface_print_via_proxy) {
+                this.pos.proxy.print_receipt(QWeb.render(
+                    'XmlReceipt', {
+                        receipt: receipt,
+                        widget: this,
+                        pos: this.pos,
+                        order: order,
+                        orderlines: order.get_orderlines(),
+                        paymentlines: order.get_paymentlines(),
+                    }));
+                this.pos.set_order(this.pos.current_order);
+                this.pos.current_order = false;
+            } else {
+                this.pos.reloaded_order = order;
+                this.gui.show_screen('receipt');
+                this.pos.reloaded_order = false;
+            }
+        },
+
+        action_return: function (order) {
+            this.pos.get('orders').add(order);
+            this.pos.set('selectedOrder', order);
+            return order;
+        },
+
+        _prepare_order_from_order_data: function (order_data) {
+            var self = this;
+            var order = new pos.Order({}, {
+                pos: this.pos,
+                temporary: !order_data.return,
+            });
+            // Set Generic Info
+            if (!order_data.return) {
+                order.name = order_data.pos_reference || order_data.name;
+            }
+            if (order_data.partner_id.length) {
+                order_data.partner_id = order_data.partner_id[0];
+            }
+            order.set_client(this.pos.db.get_partner_by_id(order_data.partner_id));
+            // Set fiscal position
+            if (order_data.fiscal_position && this.pos.fiscal_positions) {
+                var fiscal_positions = this.pos.fiscal_positions;
+                order.fiscal_position = fiscal_positions.filter(function (p) {
+                    return p.id === order_data.fiscal_position;
+                })[0];
+                order.trigger('change');
+            }
+            // Set order lines
+            var orderLines = order_data.line_ids || order_data.lines || [];
+            self._prepare_orderlines_from_order_data(order, order_data, orderLines);
+            if (order_data.return) {
+                order.return = true;
+                // A credit note should be emited if there was an invoice
+                order.set_to_invoice(order_data.origin_invoice_id);
+                // We'll refunded orders once they are synced
+                order.returned_order_id = order_data.id || order_data.name;
+                order.origin_name = order_data.pos_reference || order.returned_order_id;
+                return order;
```

Comment:
Testing more in depth the module I also seen that, if you return and order and reload the page without change anything, the "return" info is lost.
![return-pos-order](https://user-images.githubusercontent.com/4355395/57447953-91c7f500-7258-11e9-9523-2c968139be4c.gif)


A possible solution might be to add `order.trigger('change');` before return the order
```suggestion
                order.trigger('change');
                return order;
```